### ISSUE TITLE
License checklist

### ISSUE THREAD

**Issue by eboylestanley** [UBC-MDS/coRPysprofiling-R#30]

**Comment by jkim222383** [comment]
Licenses for libraries used:
- `base` R is licensed under GPLv2 | GPLv3. 
- [`here` package](https://cran.r-project.org/web/packages/here/) is licensed under MIT
- [`stopwords` package](https://cran.r-project.org/web/packages/stopwords/) is licensed under MIT
- [`tokenizers` package](https://cran.r-project.org/web/packages/tokenizers/) is licensed under MIT
- [`dplyr` package](https://cran.r-project.org/web/packages/dplyr/) is licensed under MIT

**Comment by eboylestanley** [comment]
- [`word2vec` package](https://cran.r-project.org/web/packages/word2vec/) is licensed under Apache 2
 
Data for GloVe model:
- Raw data for GloVe model was initially accessed through [`word2vec++` GitHub repository](https://github.com/maxoodf/word2vec#basic-usage), licensed under Apache 2.
 
Both `word2vec` package and `word2vec++` repository are built on the previous work by Jeffrey Pennington, Richard Socher, and Christopher D. Manning. 2014. [[GloVe: Global Vectors for Word Representation]](https://nlp.stanford.edu/pubs/glove.pdf). [[pdf]](https://nlp.stanford.edu/pubs/glove.pdf) [[bib]](https://nlp.stanford.edu/pubs/glove.bib), licensed under Apache 2.

**Comment by ssyayayy** [comment]
- [`stringr` package](https://cran.r-project.org/web/packages/stringr/) is licensed under GPLv2
- [`ggplot2` package](https://cran.r-project.org/web/packages/ggplot2/) is licensed under MIT
- [`ggwordcloud` pckage](https://cran.r-project.org/web/packages/ggwordcloud/) is licensed under GPLv3

**Comment by AnitaLi-0371** [comment]
- [`stringi` package](https://cran.r-project.org/web/packages/stringi/stringi.pdf) : 
 
> `License: The BSD-3-clause license for the package code, the ICU license for the accompanying ICU4C distribution, and the UCD license for the Unicode Character Database. See the COPYRIGHTS and LICENSE file for more details.`

**Comment by jkim222383** [comment]
MIT license is compatible with both GPLv2,3 and Apache 2.

Apache 2 is compatible with GPLv3, but the resulting work should be published under GPLv3 license, as indicated here:

> Apache 2 software can therefore be included in GPLv3 projects, because the GPLv3 license accepts our software into GPLv3 works. However, GPLv3 software cannot be included in Apache projects. The licenses are incompatible in one direction only, and it is a result of ASF's licensing philosophy and the GPLv3 authors' interpretation of copyright law. [Source](https://www.apache.org/licenses/GPL-compatibility.html)

BSD-3 ("Modified BSD liecnse") is also compatible with GPLv3:
> This is the original BSD license, modified by removal of the advertising clause. It is a lax, permissive non-copyleft free software license, compatible with the GNU GPL. [Source](https://www.gnu.org/licenses/license-list.en.html)

`stringr` package makes it a little tricky because it's licensed under GPLv2. Technically, GPLv2 is not compatible with Apache 2 or GPLv3: 

> Despite our best efforts, the FSF has never considered the Apache License to be compatible with GPL version 2, citing the patent termination and indemnification provisions as restrictions not present in the older GPL license. The Apache Software Foundation believes that you should always try to obey the constraints expressed by the copyright holder when redistributing their work. [Apache Source](https://www.apache.org/licenses/GPL-compatibility.html)

> Please note that GPLv2 is, by itself, not compatible with GPLv3. [GNU Source](https://www.gnu.org/licenses/license-list.en.html#:~:text=Please%20note%20that%20GPLv2%20is,to%20make%20the%20desired%20combination.&text=It%20is%20compatible%20with%20GPLv3.)

But GNU license page continues on to state that

> most software released under GPLv2 allows you to use the terms of later versions of the GPL as well. When this is the case, you can use the code under GPLv3 to make the desired combination. To learn more about compatibility between GNU licenses, please see our FAQ. [Source](https://www.gnu.org/licenses/license-list.en.html#:~:text=Please%20note%20that%20GPLv2%20is,to%20make%20the%20desired%20combination.&text=It%20is%20compatible%20with%20GPLv3.)

I found that all of the contributors involved with `stringr` package have agreed to change license from GPLv2 to GPLv3 in
[thread from 2016](https://github.com/tidyverse/stringr/issues/137) and [thread from 2019](https://github.com/tidyverse/stringr/issues/269). In the thread from 2019, Hadley Wickam notes that the packages licensed under GPLv2 will be updated under MIT license over time. 

Since `tidyverse` package [licensed under GPLv3](https://cran.r-project.org/web/packages/tidyverse/index.html) directly imports and uses `stringr` package licensed under GPLv2, I believe it's safe to say that `stringr` package can be used under GPLv3 license. 

In conclusion, we can publish our project under GPLv3 license, which is compatible with all of MIT license, Apache 2 license, and (in this particular case) `stringr` package.
